### PR TITLE
feature/SOPHI4-1034

### PR DESCRIPTION
**Pull request by ereasat**
Added github actions to build the project using maven

### CODE REVIEW

File: src/main/java/net/galgus/flink/streaming/connectors/http/HTTPSink.java [hunk 1]
```diff
@@ -38,6 +38,9 @@ public void invoke(IN value, Context context) throws Exception {
             writer.close();
 
             int status = conn.getResponseCode();
+            log.info("Status code: " + status);
```
Comment (aahmedg) [inline src/main/java/net/galgus/flink/streaming/connectors/http/HTTPSink.java]: change the log to log.info("Status={}", status); and add a more meaningful message

File: .github/workflows/build.yaml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+      - name: Show extracted Maven project version
+          run: echo ${{ steps.project.outputs.version }}
+
+      - name: Build with Maven
+        run: mvn clean install
+      - name: copy jar to staging
+        run: mkdir staging && cp target/*.jar staging
+      - uses: actions/upload-artifact@v2
+        with:
+          name: flink-connector-http-${{steps.project.outputs.version}}
```
Comment (aahmedg) [inline .github/workflows/build.yaml:29]: do you actually need 2 curly here?

**Review by aahmedg — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yaml (hunk 2 above)
Comment (ereasat) [reply .github/workflows/build.yaml:29]: yes, just tested it

File: src/main/java/net/galgus/flink/streaming/connectors/http/HTTPSink.java [hunk 3]
```diff
@@ -38,6 +38,11 @@ public void invoke(IN value, Context context) throws Exception {
             writer.close();
 
             int status = conn.getResponseCode();
+            log.info("Status={}", status);
+            if(status == 200) {
```
Comment (aahmedg) [inline src/main/java/net/galgus/flink/streaming/connectors/http/HTTPSink.java]: there are other successful status codes such as 201, a better check is to see if there is a method to do this already in the conn object

File: src/main/java/net/galgus/flink/streaming/connectors/http/HTTPSink.java [hunk 4]
```diff
@@ -38,6 +38,11 @@ public void invoke(IN value, Context context) throws Exception {
...
+                log.info("Message: The request has succeeded" );
```
Comment (aahmedg) [inline src/main/java/net/galgus/flink/streaming/connectors/http/HTTPSink.java]: you can merge this line with the line at 41 and just have some log messages that says, what the response code was, which url, and how long did the request take.

File: src/main/java/net/galgus/flink/streaming/connectors/http/HTTPSink.java [hunk 5]
```diff
@@ -38,6 +38,11 @@ public void invoke(IN value, Context context) throws Exception {
...
+            }
+
             if (status != 200) {
```
Comment (aahmedg) [inline src/main/java/net/galgus/flink/streaming/connectors/http/HTTPSink.java]: use if else instead.

**Review by aahmedg — CHANGES_REQUESTED** [review]

**Comment by ereasat** [comment]
The conn object does not provide any method to check successful status codes.
Also there is no method to check the request duration so I just calculated the time in the code

File: src/main/java/net/galgus/flink/streaming/connectors/http/HTTPSink.java (hunk 3 above)
Comment (ereasat) [reply src/main/java/net/galgus/flink/streaming/connectors/http/HTTPSink.java]: The conn object does not provide any method to check successful status codes. so just edited the if statement to take into account all 200 codes

File: src/main/java/net/galgus/flink/streaming/connectors/http/HTTPSink.java (hunk 4 above)
Comment (ereasat) [reply src/main/java/net/galgus/flink/streaming/connectors/http/HTTPSink.java]: there is no method to check the request duration so I just calculated the time in the code. The else statement created by the original author already logs the code, url and body for status codes other than 200, so decided not to merge with line 41

**Pull request closed without merging**